File: lua/raindrops/raindrops.lua
local rules = {
  {3, 'Pling'},
  {5, 'Plang'},
  {7, 'Plong'}
}

function raindrops(number)
  local result = ''
  for i, v in ipairs(rules) do
    n, sound = unpack(v)
    if number % n == 0 then
      result = result .. sound
    end
  end

  if result == '' then
    result = tostring(number)
  end

  return result
end

return raindrops

File: erlang/raindrops/src/raindrops.erl
-module(raindrops).

-export([convert/1, test_version/0]).

-define(Rules, [{3, "Pling"}, {5, "Plang"}, {7, "Plong"}]).

followRule(Number) ->
  fun({Factor, Sound}, Result) when Number rem Factor == 0 ->
      Result ++ Sound;
     ({_, _}, Result) ->
      Result
  end.

convert(Number) ->
  case lists:foldl(followRule(Number), "", ?Rules) of
    ""     -> integer_to_list(Number);
    Sounds -> Sounds
  end.

test_version() -> 1.
